specs/core/0_beacon-chain.md

```diff
@@ -1579,38 +1581,28 @@ The steps below happen when `state.slot > GENESIS_SLOT and (state.slot + 1) % SL
 We define epoch transition helper functions:
 
 ```python
-def get_current_total_balance(state: BeaconState) -> Gwei:
-    return get_total_balance(state, get_active_validator_indices(state, get_current_epoch(state)))
-```
-
-```python
-def get_previous_total_balance(state: BeaconState) -> Gwei:
+def get_previous_epoch_total_balance(state: BeaconState) -> Gwei:
     return get_total_balance(state, get_active_validator_indices(state, get_previous_epoch(state)))
 ```
 
-```python
-def get_unslashed_attesting_indices(state: BeaconState, attestations: List[PendingAttestation]) -> List[ValidatorIndex]:
-    output = set()
-    for a in attestations:
-        output = output.union(get_attesting_indices(state, a.data, a.aggregation_bitfield))
-    return sorted(filter(lambda index: not state.validator_registry[index].slashed, list(output)))
-```
+Note: The balance computed by `get_previous_epoch_total_balance` may be different to the actual total balance during the previous epoch transition. Due to the bounds on per-epoch validator churn and per-epoch rewards/penalties, the maximum balance difference is low and only marginally affects consensus safety.
```

Comment:
Maybe just call it `estimate_previous_epoch_total_balance`?

Reply:
The helper functions now have an `epoch` parameter which can be `current_epoch` or `previous_epoch` and we get exact effective balance :)